src/goto-instrument/contracts/contracts.cpp

```diff
@@ -294,6 +294,12 @@ void code_contractst::check_apply_loop_contracts(
   // Generate havocing code for assignment targets.
   havoc_assigns_targetst havoc_gen(to_havoc, ns);
   havoc_gen.append_full_havoc_code(loop_head_location, pre_loop_head_instrs);
+  // Add loop number to the havoc_code
+  Forall_goto_program_instructions(havoc_it, pre_loop_head_instrs)
```

Comment:
I know we have these macros, but we should really be able to use ranged `for` expressions nowadays.